src/diffusers/schedulers/schedules/flow_schedule.py

```diff
@@ -0,0 +1,183 @@
+from typing import List, Optional, Union
+
+import math
+import numpy as np
+import torch
+
+from ...configuration_utils import ConfigMixin, register_to_config
+from ..sigmas.beta_sigmas import BetaSigmas
+from ..sigmas.exponential_sigmas import ExponentialSigmas
+from ..sigmas.karras_sigmas import KarrasSigmas
+
+class FlowMatchSD3:
```

Comment:
is there any reason, the flow match sigmas are treated differently different from beta/exponential/karras? they all generate a sigma schedule, no?

Reply:
Beta/exponential/karras are more like a conversion of existing sigma schedule, the original functions are actually called `_convert_to_*`

Reply:
FlowMatchEuler currently has what we're calling FlowMatchSD3 here by default, and for other models we pass in sigmas, I think the one for SANA is different again, so it seems these will tend to be model specific, compared to everything in `BetaSchedule` like `timestep_spacing == "linspace"` `interpolation_type == "linear"` which is shared and we rarely pass in sigmas, the exception is Align Your Steps schedules which I don't think is used often. We could inline these like `timestep_spacing == "linspace"` in `BetaSchedule` but like this the idea is that we can pass a custom class that just needs a __call__ to return the sigmas if we wanted to in addition to the ones that are built-in, new models, custom schedules and experimentation can be supported easier. We could refactor `BetaSchedule` in a similar way like `LinspaceLinear` etc and allow the same level of experimentation, it just seems less likely to be used and the experimentation with this kind of scheduler will be done with conversion, there are a few more less popular from the community that we don't support but again the idea is that it can be passed in as a custom class.